app/controllers/search_controller.rb

```diff
@@ -35,14 +39,34 @@ def sort_by_priority(results, query)
     prioritized_results
   end
 
-  def sort_by_frequency(array)
-    array = array.tally
-    array = array.sort_by(&:second).reverse!
-    array.map(&:first)
+  def compare
+    lambda { |tuple1, tuple2|
+      if tuple1.second == tuple2.second
+        tuple1.third <=> tuple2.third
+      else
+        tuple2.second <=> tuple1.second
+      end
+    }
+  end
+
+  def sort_by_frequency(primary_results, seconary_results)
+    primary_results_frequencies = primary_results.tally # [[A:4][B:4][C:1]]
```

Comment:
The comment is supposed to be an example for the sorting. Do you think how the sorting works is clear without comments or should we add a more detailed example instead of this one.
For example:
<pre><code>
# Example:
# results: A, B and C
# primary result frequencies: A:4, B:4, C1
# secondary result frequencies: A:3: B:2 C:1
# primary result frequencies from A and B are the same, so the result with the smaller secondary result frequency comes first
# final sorting: B, A, C
</code></pre>